### PR TITLE
Full keyvault list

### PR DESCRIPTION
**Pull request by glenjamin**

### CODE REVIEW

File: server/services/keyVault.js [hunk 1]
```diff
@@ -197,6 +197,19 @@ async function createKeyVaultService(override) {
         };
       });
   }
+
+  async function getAllSecrets() {
```
Comment (romidane) [inline server/services/keyVault.js]: Nice

File: server/services/keyVault.js [hunk 2]
```diff
@@ -197,6 +197,19 @@ async function createKeyVaultService(override) {
...
+    const result = await client.getSecrets(keyVaultUri);
+
+    while (result.nextLink) {
```
Comment (romidane) [inline server/services/keyVault.js]: Perhaps a recursive solution such as?
```
function async collectResults(result, collection) {
  if (!result.nextLink) return collection;
  
  const next = await client.getSecretsNext(result.nextLink);
  collection.push(next);
  
  return await collectionResults(next, collection);
} 
```

Comment (glenjamin) [reply server/services/keyVault.js]: Could do, but I like the `while` loop for keeping all in one place - the eslint rule has a thing on the docs about turning it off if each loop really is dependent on the next

Comment (romidane) [reply server/services/keyVault.js]: fair enough.. it encapsulated anyway.

**Pull request merged**